apps/api_web/lib/api_web/views/shape_view.ex

```diff
@@ -2,31 +2,51 @@ defmodule ApiWeb.ShapeView do
   use ApiWeb.Web, :api_view
 
   alias ApiWeb.{RouteView, StopView}
-  alias State.{Stop, StopsOnRoute}
+  alias State.{Route, Stop, StopsOnRoute}
+
+  alias JaSerializer.Relationship.HasMany
+  alias JaSerializer.Relationship.HasOne
 
   location(:shape_location)
 
   def shape_location(shape, conn), do: shape_path(conn, :show, shape.id)
 
   attributes([:name, :direction_id, :polyline, :priority])
 
-  has_one(
-    :route,
-    type: :route,
-    serializer: RouteView
-  )
-
-  has_many(
-    :stops,
-    type: :stop,
-    identifiers: :always,
-    serializer: StopView
-  )
-
-  def stops(%{id: id, route_id: route_id}, _conn) do
-    route_id
-    |> StopsOnRoute.by_route_id(shape_ids: [id])
-    |> Stop.by_ids()
+  @impl true
+  def attributes(shape, %{assigns: %{api_version: version}} = conn)
+      when version >= "2020-04-21" do
+    shape
+    |> super(conn)
+    |> Map.take([:polyline])
```

Comment:
You can simplify this:
```
%{
  polyline: shape.polyline
}
```

Reply:
Without even calling `super`? Or are you just referring to the `Map.take/2` bit?

Reply:
All `super` really does for us here is `Map.take(struct, keys)`, so if you're building the map yourself it's easier to just do it.